src/link/inaccessible_spec_definition.rs

```diff
@@ -0,0 +1,1137 @@
+                    // validation, but results in input coercion failure for the
+                    // field at runtime.
+                    let inaccessible_super_references =
+                        implements.iter().filter_map(|interface_name| {
+                            let super_type = schema.schema().get_interface(interface_name)?;
+                            if super_type.directives.has(inaccessible_directive) {
+                                return None;
+                            }
+                            let super_field = super_type.fields.get(field_name)?;
+                            if super_field.directives.has(inaccessible_directive) {
+                                return None;
+                            }
+                            let super_argument = super_field.argument_by_name(arg_name)?;
+                            if !super_argument.directives.has(inaccessible_directive) {
+                                return None;
+                            }
+                            Some(InterfaceFieldArgumentDefinitionPosition {
+                                type_name: super_type.name.clone(),
+                                field_name: super_field.name.clone(),
+                                argument_name: super_argument.name.clone(),
+                            })
+                        });
+
+                    for inaccessible_reference in inaccessible_super_references {
+                        errors.push(SingleFederationError::RequiredInaccessible {
+                            message: format!("Argument `{inaccessible_reference}` is @inaccessible but is implemented by the argument `{type_position}.{field_name}({arg_name}:)` which is in the API schema."),
+                        }.into());
+                    }
+                }
+            }
+            validate_inaccessible_in_arguments(
+                schema,
+                inaccessible_directive,
+                match type_position {
+                    TypeDefinitionPosition::Object(object) => {
+                        HasArgumentDefinitionsPosition::ObjectField(
+                            object.field(field.name.clone()),
+                        )
+                    }
+                    TypeDefinitionPosition::Interface(interface) => {
+                        HasArgumentDefinitionsPosition::InterfaceField(
+                            interface.field(field.name.clone()),
+                        )
+                    }
+                    _ => unreachable!(),
+                },
+                &field.arguments,
+                errors,
+            )?;
+        }
+    }
+
+    if has_inaccessible_field && !has_accessible_field {
+        errors.push(SingleFederationError::OnlyInaccessibleChildren {
+            message: format!("Type `{type_position}` is in the API schema but all of its members are @inaccessible."),
+        }.into());
+    }
+
+    Ok(())
+}
+
+fn validate_inaccessible_in_values(
```

Comment:
maybe rename to `validate_inaccessible_in_enum_values`

Reply:
I've inlined it since there's only one usage site and the arguments list is almost as long as the implementation itself :)